recur/src/index.js

```diff
@@ -0,0 +1,138 @@
+import YAML from 'yaml'
+import moment from 'moment'
+import {Validator} from 'jsonschema'
+
+
+const every = {
+  id: '/Every',
+  type: 'object',
+  properties: {
+    multiplier: {
+      type: 'integer',
+      minimum: 1
+    },
+    period: {
+      type: 'string',
+      enum: ['week', 'month']
+    }
+  }
+}
+
+const on = {
+  id: '/On',
+  type: 'object',
+  properties: {
+    day: {
+      type: 'integer',
+      minimum: 1,
+      maximum: 31
+    }
+  }
+}
+
+const begin = {
+  id: '/Begin',
+  type: 'string'
+}
+
+const schema = {
+  id: '/Recur',
+  type: 'object',
+  properties: {
+    every: {'$ref': '/Every'},
+    on: {'$ref': '/On'},
+    begin: {'$ref': '/Begin'},
+  }
+}
+
+const defaults = {
+  every: {
+    multiplier: 1
+  },
+  begin: '01/01/2019'
+}
+
+
+function main() {
+
+  const now = moment() // TODO: consider timezone
+
+  getRecurConfigs()
+    .then(filterCards(now))
+    .then(updateCards)
+    .catch(err => console.log(err))
+
+  console.log({...{a: 5, b: 6}, ...{a:'a', c:5}})
+}
+
+function getRecurConfigs() {
+  // TODO: Replace with graphql query to get cards in recur list
+  const card = {
+    yaml: `
+recur:
+  every:
+    multiplier: 1
+    period: 'month'
+  on:
+    day: 1
+  begin: '01/01/2019'
+`
+  }
+  return new Promise((resolve, reject) => {
+    resolve([card])
+  })
+}
+
+function parse(yaml) {
+  const parsed = YAML.parse(yaml)
+  const withDefaults = {...defaults, ...parsed.recur}
+  const v = new Validator();
+  v.addSchema(every, '/Every')
+  v.addSchema(on, '/On')
+  v.addSchema(begin, '/Begin')
+  const result = v.validate(withDefaults, schema)
+  return {recur: withDefaults, errors: result.errors}
+}
+
+function week(begin, on, mult) {
+  return date => 
+    (on === date.day()) && (begin.week() % mult === date.week() % mult)
+}
+
+function month(begin, on, mult) {
+  return date =>
+    (on === date.date()) && (begin.month() % mult === date.month() % mult)
+}
+
+/** Create a date set with a yaml config. 
+  *
+  * A date set is a function that returns whether the given date exists in the 
+  * set.
+  *
+  */
+function dateSet(yaml) {
+  const {recur, errors} = parse(yaml)
+  const contains = {
+    week,
+    month
+  }
+  return errors.length === 0 ? 
+    contains[recur.every.period](
+      moment(recur.begin, 'DD/MM/YYYY'), 
+      recur.on.day, 
+      recur.every.multiplier 
+    ) 
+    : () => false
+}
+
+function filterCards(date) {
+  return cards => 
+    Promise.resolve(cards.filter(card => dateSet(card.yaml)(date)))
+}
+
+function updateCards(cards) {
+  // TODO: Replace with graphql query to update cards
```

Comment:
What are we updating?

Reply:
Would this be clearer?
```suggestion
  // TODO: Replace with graphql query to create new cards for each recurring one
```
Which actually brings up a point that we might've been forgetting - the user config (envs?) should also contain the name of the list in which these new cards get pushed to.

Reply:
That's correct, but I would argue it won't be the .env since they might want to configure or change it conveniently. 
The alternative would be to include the destination list (name or id perhaps) in the yaml